src/Symfony/Bridge/Doctrine/Form/DoctrineOrmTypeGuesser.php

```diff
@@ -56,31 +57,32 @@ public function guessType($class, $property)
             return new TypeGuess('Symfony\Bridge\Doctrine\Form\Type\EntityType', ['em' => $name, 'class' => $mapping['targetEntity'], 'multiple' => $multiple], Guess::HIGH_CONFIDENCE);
         }
 
+        $useDeprecatedConstants = !class_exists(Types::class);
```

Comment:
Maybe you could store the class in this constant : `$classTypes = class_exists(Types::class) ? Types::class : Type::class;` and then call `$classTypes::...` ? If I remember correctly it's available in PHP 5.5+ (Otherwise, `constant(sprintf(...))` could do the trick)